include/multipass/cloud_init_iso.h

```diff
@@ -32,10 +33,21 @@ class CloudInitIso
 public:
     void add_file(const std::string& name, const std::string& data);
     void write_to(const Path& path);
+    void read_from(const std::filesystem::path& path);
+
+    friend bool operator==(const CloudInitIso& lhs, const CloudInitIso& rhs)
+    {
+        return lhs.files == rhs.files;
+    }
```

Comment:
I think the tiny difference between member function overload and friend function overload to client code is that the member function overload may not be able to implicitly convert left side value to the class type when the provided value requires implicit conversion but the right side can which does not conforms to symmetry. However, that is a general statement, in this particular class, there is no implicit conversion involved constructor so it would be the same to the client code. 

But visually, the friend function overload signature has both left side and right side data as input arguments, which highlights the symmetry nature of the operator.

Reply:
Ok, makes sense. We'll keep the friend function. Thanks for the explanation!